packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -442,225 +500,216 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 					deviceId,
 					deviceOptions,
 					getCurrentTime,
 					threadedClassOptions
 				)
-			} else if (deviceOptions.type === DeviceType.TELEMETRICS) {
-				newDevice = await DeviceContainer.create<DeviceOptionsTelemetrics, typeof TelemetricsDevice>(
-					'../../dist/integrations/telemetrics/index.js',
-					'TelemetricsDevice',
+			case DeviceType.TRICASTER:
+				return DeviceContainer.create<DeviceOptionsTriCasterInternal, typeof TriCasterDevice>(
+					'../../dist/integrations/tricaster/index.js',
+					'TriCasterDevice',
 					deviceId,
 					deviceOptions,
 					getCurrentTime,
 					threadedClassOptions
 				)
-			} else {
-				// @ts-ignore deviceOptions.type is of type "never"
-				const type: any = deviceOptions.type
-				return Promise.reject(`No matching device type for "${type}" ("${DeviceType[type]}") found in conductor`)
-			}
-
-			if (!newDevice) {
-				const type: any = deviceOptions.type
-				return Promise.reject(`No device could be created for "${type}" ("${DeviceType[type]}")`)
-			}
-
-			newDevice.device.on('resetResolver', () => this.resetResolver()).catch(console.error)
-			newDevice.on('error', (context, e) => {
-				this.emit('error', `deviceContainer for "${newDevice?.deviceId}" emitted an error: ${context}, ${e}`)
-			})
-
-			// Double check that it hasnt been created while we were busy waiting
-			if (this.devices.has(deviceId)) {
-				throw new Error(`Device "${deviceId}" already exists when creating device`)
-			}
-			this.devices.set(deviceId, newDevice)
-
-			return newDevice
-		} catch (e) {
-			if (newDevice) {
-				try {
-					await newDevice.terminate()
-				} catch (e) {
-					this.emit('error', `Cleanup failed of aborted device "${newDevice.deviceId}": ${e}`)
-				}
-			}
-			this.devices.delete(deviceId)
-			this.emit('error', 'conductor.createDevice', e)
-			return Promise.reject(e)
+			default:
+				assertNever(deviceOptions)
+				return null
 		}
 	}
+
+	private async terminateUnwantedDevice(newDevice: DeviceContainer<DeviceOptionsBase<any>> | undefined) {
+		await newDevice
+			?.terminate()
+			.catch((e) => this.emit('error', `Cleanup failed of aborted device "${newDevice.deviceId}": ${e}`))
+	}
+
 	/**
 	 * Initialises an existing device that can be referenced by the timeline and mappings.
 	 * @param deviceId Id used by the mappings to reference the device.
 	 * @param deviceOptions The options used to initalize the device
+	 * @param activeRundownPlaylistId Id of the current rundown playlist
+	 * @param options Additional options
 	 * @returns A promise that resolves with the initialised device, or rejects with an error message.
 	 */
 	public async initDevice(
 		deviceId: string,
 		deviceOptions: DeviceOptionsAnyInternal,
-		activeRundownPlaylistId?: string
+		activeRundownPlaylistId?: string,
+		options?: { signal?: AbortSignal }
 	): Promise<DeviceContainer<DeviceOptionsBase<any>>> {
+		const throwIfAborted = this.throwIfAborted.bind(this, deviceId, 'initialisation')
```

Comment:
```suggestion
		const throwIfAborted = this.throwIfAborted.bind(this, options?.signal, deviceId, 'creation')
```

Missing a parameter. Functions like `.bind` and `.call` are not being type safe, but should be enabled in our config https://www.typescriptlang.org/tsconfig#strictBindCallApply

Reply:
Oh, by 'strict' they mean this signature `bind<T, AX, R>(this: (this: T, ...args: AX[]) => R, thisArg: T, ...args: AX[]): (...args: AX[]) => R;` Which this call does fall into as the arguments do match `(string | AbortController | undefined)[]`.

So yeah, I don't like the use of `.bind()` here. I would rather see this as `const throwIfAborted = ( () => this.throwIfAborted(options?.signal, deviceId, 'initialisation')` so that we have real type safety